hypershift-operator/metrics.go

```diff
@@ -178,7 +202,36 @@ func (m *hypershiftMetrics) observeHostedClusters(hostedClusters *hyperv1.Hosted
 	hcCount := newLabelCounter()
 	hcByConditions := newLabelCounter()
 
+	identityProvidersCounter := make(map[string]float64)
 	for _, hc := range hostedClusters.Items {
+		// Collect proxy metric.
+		var proxyHTTP, proxyHTTPS, proxyTrustedCA string
+		if hc.Spec.Configuration != nil && hc.Spec.Configuration.Proxy != nil {
+			if hc.Spec.Configuration.Proxy.HTTPProxy != "" {
+				proxyHTTP = "1"
+			}
+			if hc.Spec.Configuration.Proxy.HTTPSProxy != "" {
+				proxyHTTPS = "1"
+			}
+			if hc.Spec.Configuration.Proxy.TrustedCA.Name != "" {
+				proxyTrustedCA = "1"
+			}
+			m.clusterProxy.WithLabelValues(hc.Namespace, hc.Name, proxyHTTP, proxyHTTPS, proxyTrustedCA).Set(1)
```

Comment:
@enxebre my understanding is that all metrics should be given a value (whether something is enabled or not). In the case of proxy, if proxy is not configured, you need to set the value to 0.

Also my understanding of the label values is that they're supposed to have the value of the setting and not 1/0 whether they're set:
https://github.com/openshift/osd-metrics-exporter/blob/be64e94da0caa0ba7941324084b4b14ba97c70c6/pkg/metrics/aggregator.go#L175-L181

Reply:
>Also my understanding of the label values is that they're supposed to have the value of the setting and not 1/0 whether they're set:

they are actually setting 1/0 https://github.com/openshift/osd-metrics-exporter/blob/d834f20ed0a2314a0bda8b8591dc2b20df1f6251/controllers/proxy/proxy_controller.go#L72-L93

>@enxebre my understanding is that all metrics should be given a value (whether something is enabled or not). In the case of proxy, if proxy is not configured, you need to set the value to 0.

I see. Is that a recommended practice or is there any other reason? 0 vs none